### DISCUSSION TITLE
Migrating Query.get -> Session.get, what to do with Query.options?

### DISCUSSION THREAD

**Discussion by dkbarn** [sqlalchemy/sqlalchemy#10881] [Usage Questions] [answered]
I'm in the midst of my journey migrating from SQLAlchemy 1.4 to 2.0, following the migration plan and dealing with all of the deprecation warnings. At the moment I'm dealing with updating all `Query.get` calls to be `Session.get` instead. It's unclear to me, and not covered by the documentation, what you do with any non-trivial `Query.get` calls, e.g. ones that include `Query.options` for eager loading of relationships.

For example, how would I migrate this to be 2.x compliant?
```
instance = (
    session.query(MediaInstance)
        .options(
            joinedload(MediaInstance.photo_master),
            joinedload(MediaInstance.media),
        )
        .get(instance_id)
)
```

**Comment by CaselIT** [comment, marked as answer]
Hi,

This is documented. Get accepts options: https://docs.sqlalchemy.org/en/20/orm/session_api.html#sqlalchemy.orm.Session.get.params.options

**Reply by dkbarn** [reply]
Thank you @CaselIT 🙏 I definitely missed that.

So the 2.x compliant version would be:
```
instance = session.get(
    MediaInstance,
    instance_id,
    options=[
        joinedload(MediaInstance.photo_master),
        joinedload(MediaInstance.media),
    ]
)
```

**Reply by CaselIT** [reply]
yes, that's the idea!
